DNN Platform/Website/DesktopModules/Admin/Authentication/Login.ascx.cs

```diff
@@ -796,10 +798,21 @@ private static string ReplaceLanguage(string Url, string originalLanguage, strin
             return returnValue;
         }
 
+        private string GetLoginPath()
+        {
+            if (this.PortalSettings.LoginTabId == Null.NullInteger)
+            {
+                return LOGIN_PATH;
+            }
+
+            var tab = TabController.Instance.GetTab(this.PortalSettings.LoginTabId, this.PortalId);
+            return tab != null ? $"/{tab.TabName}" : LOGIN_PATH;

```

Comment:
I don't think `$"/{tab.TabName}"` is going to work for all scenarios (e.g. what if the page name has a space in it).  The URL should be generated via `INavigationManager.NavigateURL` and then compare its `LocalPath` to `UrlReferrer.LocalPath`

Reply:
LocalPath EndsWith comparison not matching

![image](https://user-images.githubusercontent.com/7112371/95978672-1ea4b400-0e38-11eb-801e-1eeaf6354b37.png)

Reply:
this.Request.UrlReferrer.LocalPath EndsWith comparison of GetLoginPath() fails whereas GetLoginPath() StartsWith comparison of this.Request.UrlReferrer.LocalPath succeeds. Updated the comparison and verified the fix